packages/expo-background-fetch/ios/EXBackgroundFetch/EXBackgroundFetch.m

```diff
@@ -0,0 +1,120 @@
+// Copyright 2018-present 650 Industries. All rights reserved.
+
+#import <EXCore/EXDefines.h>
+#import <EXBackgroundFetch/EXBackgroundFetch.h>
+#import <EXBackgroundFetch/EXBackgroundFetchConstants.h>
+#import <EXBackgroundFetch/EXBackgroundFetchTaskConsumer.h>
+#import <EXTaskManagerInterface/EXTaskManagerInterface.h>
+
+// Background Fetch statuses (equivalents to UIBackgroundRefreshStatus)
+NSString *const EXBackgroundFetchStatusRestricted = @"restricted";
+NSString *const EXBackgroundFetchStatusDenied = @"denied";
+NSString *const EXBackgroundFetchStatusAvailable = @"available";
+
+// Possible results (equivalents to UIBackgroundFetchResult)
+NSString *const EXBackgroundFetchResultNoData = @"no-data";
+NSString *const EXBackgroundFetchResultNewData = @"new-data";
+NSString *const EXBackgroundFetchResultFailed = @"failed";
+
+@interface EXBackgroundFetch ()
+
+@property (nonatomic, weak) id<EXTaskManagerInterface> taskManager;
+
+@end
+
+@implementation EXBackgroundFetch
+
+EX_EXPORT_MODULE(ExpoBackgroundFetch);
+
+- (NSDictionary *)constantsToExport
+{
+  return @{
+           @"Status": @{
+               @"RESTRICTED": EXBackgroundFetchStatusRestricted,
+               @"DENIED": EXBackgroundFetchStatusDenied,
+               @"AVAILABLE": EXBackgroundFetchStatusAvailable,
+               },
+           @"Result": @{
+               @"NO_DATA": EXBackgroundFetchResultNoData,
+               @"NEW_DATA": EXBackgroundFetchResultNewData,
+               @"FAILED": EXBackgroundFetchResultFailed,
+               }
+           };
+}
+
+- (void)setModuleRegistry:(EXModuleRegistry *)moduleRegistry
+{
+  _taskManager = [moduleRegistry getModuleImplementingProtocol:@protocol(EXTaskManagerInterface)];
+}
+
+EX_EXPORT_METHOD_AS(getStatusAsync,
+                    getStatus:(EXPromiseResolveBlock)resolve
+                    reject:(EXPromiseRejectBlock)reject)
+{
+  resolve([self _getStatus]);
+}
+
+EX_EXPORT_METHOD_AS(setMinimumIntervalAsync,
+                    setMinimumInterval:(nonnull NSNumber *)minimumInterval
+                    resolve:(EXPromiseResolveBlock)resolve
+                    reject:(EXPromiseRejectBlock)reject)
+{
+  NSTimeInterval timeInterval = [minimumInterval doubleValue];
+  [[UIApplication sharedApplication] setMinimumBackgroundFetchInterval:timeInterval];
+  resolve([NSNull null]);
```

Comment:
Since we can't resolve promises to `undefined` on Android, I think it would be a good idea to always return `null` to JavaScript, so it's always clear how to type the return value in TypeScript (`Promise<null>`). Also, it might be confusing that it returns `undefined` on iOS and `null` on Android 😅 So I just wanted to make it consistent.